scripts/policy/frameworks/cluster/controller/main.zeek

```diff
@@ -475,6 +463,84 @@ event ClusterController::API::get_instances_request(reqid: string)
 	event ClusterController::API::get_instances_response(reqid, res);
 	}
 
+event ClusterAgent::API::get_nodes_response(reqid: string, result: ClusterController::Types::Result)
+	{
+	ClusterController::Log::info(fmt("rx ClusterAgent::API::get_nodes_response %s", reqid));
+
+	# Retrieve state for the request we just got a response to
+	local areq = ClusterController::Request::lookup(reqid);
+	if ( ClusterController::Request::is_null(areq) )
+		return;
+
+	# Release the request, which is now done.
+	ClusterController::Request::finish(areq$id);
+
+	# Find the original request from the client
+	local req = ClusterController::Request::lookup(areq$parent_id);
+	if ( ClusterController::Request::is_null(req) )
+		return;
+
+	# Zeek's ingestion of an any-typed val via Broker yields an opaque
+	# Broker DataVal. When Zeek forwards this val via another event it stays
+	# in this opaque form. To avoid forcing recipients to distinguish
+	# whether the val is of the actual, intended (any-)type or a Broker
+	# DataVal wrapper, we explicitly cast it back to our intended Zeek
+	# type. This test case demonstrates: broker.remote_event_vector_any
+	result$data = result$data as ClusterController::Types::NodeStatusVec;
```

Comment:
This bit tripped me up pretty significantly. It would be nice if we could keep `any`-typed values more transparent. I shouldn't have to think about whether this value might result from a Broker-borne event or have been created locally.

Reply:
I believe that's a work-around for otherwise not being able to transmit type information for the `any` value, i.e., we wouldn't know how to reconstruct it on the receiver side. But yeah, not great. We should be able to do that unwrapping behind the scenes at the receiver I'd think.